bluemira/base/components.py

```diff
@@ -189,7 +201,7 @@ def copy_children(
 
     def get_component(
         self, name: str, first: bool = True, full_tree: bool = False
-    ) -> Union[Component, Tuple[Component], None]:
+    ) -> Union[Component, Iterable[Component], None]:
```

Comment:
I think we should be as specific as possible when it comes to return types, and not use general types like `Iterable`. This function is a bit confusing, though. It sometimes returns a list and sometimes a tuple. 🤔  

In my ideal world, this would _always_ return a list. I think it puts an undue burden on the caller to need to check their return types every time they use the function. Not to mention how much our use of this function upsets MyPy.

Back to reality, though, my take is that we should revert the change to this line, and update [line 267](https://github.com/Fusion-Power-Plant-Framework/bluemira/pull/2236/files#diff-4557c02eb70464b83b973f6f28f0ae18a4ce0b351f00dddbb96c3c1acedc5cc2L264) to return a tuple (technically an API change, but we can probably get away with it). Thoughts?

Reply:
It always returns a tuple at the moment (I just was mucking about with stuff..). 

`get_component_properties` would need to change L284 not L276 if you wanted to return lists but thats a bit unrelated to this change. 
That function at the moment returns lists for the grouping of a selected property and a tuple of there is more than 1 property, the same as the normal return of any function that has multiple returns (it also returns a single property if that is all thats found).

Reply:
Ah, sorry, I think I was getting a bit confused between this function and `get_component_properties`, my bad.